lib/model.py

```diff
@@ -6,25 +6,25 @@
 from enum import Enum
 from lib.timer import Timer, msec, seconds, sec_str, to_msec, to_seconds, years
 from lib.config import Configuration
-from typing import Any
 from collections import defaultdict
+from lib.types import UserProfileType, ChallengeType, GameEventType, GameStateType, PlayerType
 
 logger = logging.getLogger(__name__)
 
 
 class Challenge:
     """Store information about a challenge."""
 
-    def __init__(self, challenge_info: dict[str, Any], user_profile: dict[str, Any]) -> None:
+    def __init__(self, challenge_info: ChallengeType, user_profile: UserProfileType) -> None:
         """:param user_profile: Information about our bot."""
         self.id = challenge_info["id"]
         self.rated = challenge_info["rated"]
         self.variant = challenge_info["variant"]["key"]
         self.perf_name = challenge_info["perf"]["name"]
         self.speed = challenge_info["speed"]
-        self.increment: int = challenge_info.get("timeControl", {}).get("increment")
-        self.base: int = challenge_info.get("timeControl", {}).get("limit")
-        self.days: int = challenge_info.get("timeControl", {}).get("daysPerTurn")
+        self.increment: int = challenge_info.get("timeControl", {}).get("increment", -1)
+        self.base: int = challenge_info.get("timeControl", {}).get("limit", -1)
+        self.days: int = challenge_info.get("timeControl", {}).get("daysPerTurn", -1)
```

Comment:
Why is -1 used as a default value here? This conflicts with the check in `is_supported_time_control()` below, which checks for `None`. I think changing the type of these members to `Optional[int]` is better.